open-apparel-registry/open-apparel-registry
# Ignore partial OAR ids

## Overview

When searching for a number, such as '12', in the OAR text search, all
facilities with the number '12' in their OAR ID will be returned. In
general, users who want to match to an OAR ID will search for the
entire ID, and not just a portion of the ID. This results in many more
facilities being returned for numeric searched than users are expecting.

We now only consider the query text to be a match to the OAR ID if it's
an exact match.

Working with exact matches revealed a related bug. 
When the search is run for the facilities list, if there is only one
match, the search bar automatically navigates to the facility. However,
when we navigate back to the list from the facility details, this
resulted in being navigated directly back to the facility details.

When rerunning the search in the back navigation for facility details,
replace the 'push' command with a noop to prevent navigation to a
specific facility.

Connects #1640 

## Demo

Results when searching for '5':
<img width="359" alt="Screen Shot 2022-02-11 at 11 27 59 AM" src="https://user-images.githubusercontent.com/21046714/153658100-a0e6fa54-8f3d-4b0e-9268-db61a027e842.png">

## Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

- On `develop`:
    - Search for ‘5’ in the text search. You will find over 100 facilities. 
    - Navigate to a facility and copy the OAR ID. Enter the OAR ID in the text search and search. 
    - You will be navigated automatically to a matching facility. Click the back button, but you will not be navigated back to the facilities list.
- On this branch: 
    - Search for ‘5’ in the test search. You will now only find a few facilities (all with ‘5’ in their name).
    - Search for the same OAR ID as before. 
    - You will be navigated automatically to the matching facility. Click the back button, and you will be navigated to the facility list.
    - Repeat the search in an embedded map and confirm that general text search functionality behaves as before. 

## Checklist
